crates/primitives/src/chain_spec.rs

```diff
@@ -37,7 +37,13 @@ pub trait TransactionValidation {
     }
 }
 
-pub trait EvmWiring: Sized + 'static {
+
+pub trait EvmWiring: Sized {
+    /// External type
+    type ExternalContext: Sized;
+    /// Database type.
+    type Database: Database;
+
```

Comment:
It seems to me that three variability concepts are being intertwined here:
1. An EVM can receive different `Database` implementations
2. An EVM can receive different `ExternalContext` data (e.g. an inspector)
3. DIfferent chain types (L1 Ethereum vs Optimism vs Arbitrum) have their own implementation of EVM types (transaction, block, etc.)

By tying the 1st and 2nd form of variability, it now becomes impossible to reuse the same implementation of of L1 Ethereum or Optimism for differing `Database` implementations or `ExternalContext` data.

Is what you want to do here something along the lines of:

```rs
pub trait EvmWiring {
    /// External type
    type ExternalContext: Sized;
    /// Database type.
    type Database: Database;
    /// Chain spec
    type ChainSpec: ChainSpec;
}
```

Even this feels like it's creating unnecessary boilerplate as you'll need to create a different `EvmWiring` type for each combination of the three variabilities.

To me it seems that each of these should be its own generic type on the EVM struct(s)

Reply:
I added a `EthereumWiring<DB,EXT>` this would allow reusage of same wire, even switching database inside this repo would not be possible without this.

```
#[derive(Clone, Copy, Debug, Default, PartialEq, Eq, Hash, PartialOrd, Ord)]
pub struct EthereumWiring<DB: Database, EXT> {
    phantom: core::marker::PhantomData<(DB, EXT)>,
}
```